presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -140,6 +140,10 @@ public Session createSession(QueryId queryId, SessionContext context, Optional<S
                 .setTraceToken(context.getTraceToken())
                 .setResourceEstimates(context.getResourceEstimates());
 
+        if (context.getPath() != null) {
+            sessionBuilder.setPath(new SqlPath(Optional.of(context.getPath())));
```

Comment:
According to `SessionContext` interface `getPath` is `Nullable`, so we should call `Optional.ofNullable(context.getPath())`.

Reply:
Even after the check to make sure that it's not null?

Reply:
Oops I missed that, you are right.